### ISSUE TITLE
We should keep our roadmap updated

### ISSUE THREAD

**Issue by choldgraf** [jupyter/roadmap#43]
As a part of the numfocus summit @rgbkrk rightly pointed out that it's a shame we don't keep this repository updated! Is there a reason it isn't currently kept up-to-date? I think it could be useful to do so (and I'm happy to put in time trying to update the projects that I help out with).

Conversely - if we're *not* going to update a lot of this documentation, perhaps we should consider sunsetting some of these? I feel like it almost looks *worse* if we have roadmaps that aren't updated at all (e.g. over the course of 2 years), vs roadmaps that aren't visible.

Thoughts?

**Comment by SylvainCorlay** [comment]
We will certainly update the ipywidgets one. Thanks for the heads up.

**Comment by jasongrout** [comment]
Having a roadmap separate from the actual repo makes it hard to remember to keep up to date (out of sight, out of mind...). Perhaps instead we should have a project convention, perhaps a ROADMAP.md file in the root of the repo? That way the roadmap is closer to the code.

**Comment by choldgraf** [comment]
ROADMAP.md is a clever idea. I'd be +1 on this. Curious what @willingc and @SylvainCorlay 
 @Ruv7 and @ivanov think about this as I think as they were chatting roadmaps at the summit today too!

**Comment by rgbkrk** [comment]
I'm cool with creating a `ROADMAP.md` per project and calling attention to it from this repository.

The only roadmap I've worked on in here is cross-repository and project `spark.md`, since it touches frontends, kernels, jupyter client, nbformat, and spark itself. I could just make a separate repo for tracking that work. 😄

**Comment by blink1073** [comment]
We could leave this repo here for cross-cutting roadmaps and link out to repo-specific roadmaps where appropriate.

**Comment by willingc** [comment]
While I think `ROADMAP.md` per project is a fine idea, I believe that serves a different purpose than what funders were interested in seeing. 

They were looking for a roadmap for each project in a similar format. I think to make these useful to them that they be located in the same place (which could be a repo or static site page).

**Comment by blink1073** [comment]
An alternative is to make updating the roadmap part of a release checklist, and always copy-paste release checklists into github issues to make sure they are all addressed.

**Comment by jasongrout** [comment]
Perhaps a new issue template for release issues would be appropriate?

**Comment by choldgraf** [comment]
@willingc I agree re: funders. Though I'd love to see roadmaps being used as a way of generating excitement and interest in contributing from others in the broader community. Maybe we will get people interested in helping out with a feature if they know we're interested in seeing it happen.

One thing we could do is have a page on `jupyter.org` that links out to some specific project roadmap files (or we could aggregate them with a script into a single document, though this might be a bit complex to organize).

**Comment by blink1073** [comment]
> Perhaps a new issue template for release issues would be appropriate?

I'd worry that it would be a confusing option for new contributors.

**Comment by choldgraf** [comment]
@blink1073 - it looks like we *could* use an issue template by putting it in a `.github` folder and specifying it in the URL:

https://blog.github.com/2018-01-25-multiple-issue-and-pull-request-templates/

It'd be something that release managers would need to "know" to do...so in that case maybe it wouldn't be much more useful than just having a centralized markdown file that can be copy-pasted by maintainers in case it's useful. What do you think?

**Comment by willingc** [comment]
I may be wrong but I thought the issue templates display whenever a user clicks new issue. This may be as @blink1073 mentions confusing to users.

Possibly having a simple starter template in markdown would be fine.

**Comment by blink1073** [comment]
Thanks @willingc, that captures the thought.

**Comment by choldgraf** [comment]
So, I'm not sure if there's anything particularly *actionable* about this issue, in that there's nothing that can require anybody to do anything here re: roadmaps :-)

However, what I can commit to is bringing up the idea of creating a `ROADMAP.md` file in one or multiple of the JupyterHub repos at our next JHub team meeting, and we'll see how that goes. I'd be great if the other neighborhoods in Jupyter land gave it a shot too (if you think it'll be useful)

**Comment by pratiksha-chavan10** [comment]
Hi, I am a beginner in data science.
I would like to contribute. Can you guide me on how to help with this?
